cl/phase1/network/services/attestation_service.go

```diff
@@ -0,0 +1,149 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ledgerwatch/erigon/cl/beacon/synced_data"
+	"github.com/ledgerwatch/erigon/cl/clparams"
+	"github.com/ledgerwatch/erigon/cl/cltypes/solid"
+	"github.com/ledgerwatch/erigon/cl/phase1/forkchoice"
+	"github.com/ledgerwatch/erigon/cl/phase1/network/subnets"
+	"github.com/ledgerwatch/erigon/cl/utils/eth_clock"
+	"github.com/ledgerwatch/erigon/cl/validator/committee_subscription"
+)
+
+type attestationService struct {
+	forkchoiceStore    forkchoice.ForkChoiceStorage
+	committeeSubscribe *committee_subscription.CommitteeSubscribeMgmt
+	ethClock           eth_clock.EthereumClock
+	syncedDataManager  *synced_data.SyncedDataManager
+	beaconCfg          *clparams.BeaconChainConfig
+	netCfg             *clparams.NetworkConfig
+	// validatorAttestationSeen maps from epoch to validator index. This is used to ignore duplicate validator attestations in the same epoch.
+	validatorAttestationSeen map[uint64]map[uint64]struct{}
+	validatorAttSeenLock     sync.Mutex
+}
+
+func NewAttestationService(
```

Comment:
Try to write unit tests for this service you can see how i did in block_service_test.go